### PR TITLE
Let teacher request ai eval

### PR DESCRIPTION
**Pull request by davidsbailey**
Finishes https://codedotorg.atlassian.net/browse/AITT-101. This PR does the following:
* adds a new `/rubrics/1/ai_evaluation_status_for_user` endpoint, which returns boolean status variables for whether the student has `attempted` the current level, and whether that attempt has been evaluated (`lastAttemptEvaluated`). 
  * `attempted` is computed by looking for presence of a UserLevel
  * `lastAttemptEvaluated` is computed by comparing `user_level.updated_at` to the `update_at` timestamp on the most recent `LearningGoalAIEvaluation`
* updates `/rubrics/1/run_ai_evaluation_for_user` to reject the request unless the level has been attempted but not yet evaluated
* updates the UI to display a message according to this status, following the state transition diagram below
* adds unit tests for ruby and JS

Kudos to Bethany for adding the UI messages and writing the JS tests!

### state transition diagram

Here is how the client state behaves in response to the AI evaluation status and user clicks (apologies, I left out EVALUATION PENDING which happens after the run ai button is clicked):

PXL_20231024_223640855 MP

### demo

this screen capture shows the following states: INITIAL_LOAD (very brief) --> NOT_ATTEMPTED --> student attempts --> teacher refreshes --> INITIAL_LOAD --> READY --> teacher clicks run --> EVALUATION_PENDING --> SUCCESS

https://github.com/code-dot-org/code-dot-org/assets/8001765/7efb9be2-2f9f-4a32-a95c-79b3085fb493

## Testing story

* new js tests
* new and updated ruby tests

### CODE REVIEW

File: apps/test/unit/templates/rubrics/RubricSettingsTest.jsx
Comment (davidsbailey) [inline apps/test/unit/templates/rubrics/RubricSettingsTest.jsx:1]: These tests look great Bethany, thank you so much for pitching in! LGTM on changes to this file

File: apps/src/templates/rubrics/RubricSettings.jsx
Comment (davidsbailey) [inline apps/src/templates/rubrics/RubricSettings.jsx:1]: LGTM on Bethany's changes to this file

**Review by wilkie — APPROVED** [review]
Looks good and I'm definitely appreciative of these good tests!

**Review by bethanyaconnor — APPROVED** [review]
Thanks for thinking through this so deeply!

**Pull request merged**